lib/model/src/main/java/io/atlasmap/v2/Field.java

```diff
@@ -437,4 +439,15 @@ public String getName() {
     public void setName(String value) {
         this.name = value;
     }
+
+
+    /**
+     * A shortcut to get fields of a collection as a list.
+     *
+     * @return list of fields
+     */
+    @JsonIgnore
+    public List<? extends Field> getCollectionFields() {
```

Comment:
What kind of job will `getCollectionFields()` do? currently it looks like only returning children in the complex field?

Reply:
FYI, `Collection Field` in AtlasMap is a special term for the array/list field, like `/employee<>/email<>`

Reply:
Yes, it's a convenient method to determine if the field has any children. Otherwise I'd need to check for an implementation, cast and get children. It allows to abstract that out.

Reply:
Maybe rename it to getChildrenFields() then?

Reply:
Hmm OK, that's a downside of having complex field classes (*ComplexType) separated from the primitive fields (*Field).  Although we might be able to clean them up in next major version, I'd like to avoid returning null for this from the primitive fields as its intention is not clear at a glance, and it looks confusing as we already have complex field types. Can we also move it as a static utility method?

Reply:
I made getChildrenFields return empty list by default. Not sure about the static utility method. I would still need to cast to different types in that method and make it aware of all implementations. Where should I put it?

Reply:
Maybe to AtlasUtil just like `excludeNotRequestedFields()`? with same reason, keep model objects from having a logic other than getter/setter.
We can avoid casting to the module specific complex type like `JsonComplexType` by adding an interface, e.g. `ComplexType` which only declares the getter/setter for the children. Does it help?

Reply:
If we are to add an interface to ComplexTypes then there's no need for a util method as it's simply a getter. I'm okay with doing that instead of adding it to the Field.

Reply:
Sounds good :+1:

Reply:
Done!